src/components/Tooltip/BaseTooltip/index.tsx

```diff
@@ -184,6 +185,18 @@ function Tooltip(
         setIsVisible(false);
     }, []);
 
+    const child = useMemo(() => getFirstValue(children), [children]);
```

Comment:
This line ensures that the children is indeed a single value, which avoids [app crashing when strikethrough is used](https://github.com/Expensify/App/issues/32379)

CC @cubuspl42

Reply:
Dejavu.

![image](https://github.com/Expensify/App/assets/2590174/aaac746b-6b89-42be-bbdf-3ed0b7947e12)

If I remember correctly, this is done under an assumption that `children` can only be a single-element parent, but this assumption is neither checked nor commented on.

I remember that I suggested `Children.only` but you explained that it doesn't suit this case. I also asked whether the usages can be fixed, so this code is not necessary at all.

Reply:
Oh, I think we had to misunderstood each other - this is the _use-case_ of `Hoverable` where we want to narrow its children to a single child. The typings are what is constraining the children (the type used is `ReactElement`), but there are still `.js` files that do not obey it.

In order to find what is going on, I've analyzed usages of `Tooltip` component but there were none where multiple children are passed. I would be grateful if you could double check that so we are sure that if escape hatch is needed or not. Thanks a lot!

Reply:
Yes, I got confused. This is a different place.

Reply:
So, restarting the thread:

Isn't _this_ place suitable for `Children.only`?

Reply:
@kacper-mikolajczak From my perspective, this is the important thread.

> I would be grateful if you could double check 

I will, during the review. Unfortunately, before that, you'll have to analyze it yourself and take responsibility (which comes with great power, which programming is).

<img src="https://media0.giphy.com/media/wkTqDgOILTHFK/giphy.gif"/>

Reply:
> Isn't this place suitable for Children.only?

It's not, `Children.only` always throws when array is received (taken from [docs](https://react.dev/reference/react/Children#children-only)):

> This method always throws if you pass an array (such as the return value of Children.map) as children. In other words, it enforces that children is a single React element, not that it’s an array with a single element.

In general I agree with you, that's why I went through the usages and managed to find the [fix](https://github.com/Expensify/App/pull/33257/commits/0af0d78214fa12e9c0bfdac97864da696895d039). In order to remove the clunky

```jsx
const child = useMemo(() => getFirstValue(children), [children]);
```

an adjustment in `strikethrough` logic of `OfflineWithFeedback` was needed - when we are mapping the children, instead of always returning the array, we will return child directly if it is only child. 

I find it the most suitable approach so far - let me know what you think 👍